core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -303,7 +308,8 @@ private void updateHealthCheckedState(SubchannelData subchannelData) {
     if (subchannelData.state != READY) {
       return;
     }
-    if (subchannelData.getHealthState() == READY) {
+
+    if (notAPetiolePolicy || subchannelData.getHealthState() == READY) {
```

Comment:
You can just always set health status to READY when creating the subchannel. Same as L454.

Reply:
I've updated the if at line 453 to use notAPetiolePolicy for auto setting as ready, but I'd like to leave this here to make it clear to people maintaining the code what is happening since understanding that somewhere else set the value to READY and nowhere else should have changed it is difficult.